pytest_splunk_addon/standard_lib/sample_generation/sample_event.py

```diff
@@ -301,6 +301,6 @@ def update_metadata(self, event, metadata, key_fields):
 
             return event, metadata, key_fields
 
-        except IndexError as error:
+        except KeyError as error:
```

Comment:
The only change made in source code during unit tests implementation

Reply:
I believe that this should be fine since it is checking a dictionary object.